### DISCUSSION TITLE
Extension to environment variable support

### DISCUSSION THREAD

**Discussion by DaveDavenport** [davatorium/rofi#1411] [Ideas/Feature Requests]
I am pondering to allow the following construct in a theme:

```css
window {
  width: ${NEW_WIDTH|"80%"};
}
```

So  if you  run it normally the width is 80%, but if you run:

```bash
NEW_WIDTH="50px" rofi -show run
```
the width is 500px.
Would this be  a usable feature?    This would allow for easily tweaking of any of the properties.

**Comment by khlsvr** [comment]
I brought this up yesterday on the IRC for having my rofi updated to the new config I found out the -width parameter on command line didn't work. I had used different width for a few things, but they are not that important at least for now so I can survive without this feature. If this feature existed, I would probably use it as I would only need to modify two files I guess. It's not as "good" as editing one file with the -width parameter though.

**Reply by DaveDavenport** [reply]
For me the width is part of the theme, not something you dynamically modify, but I saw your usecase so that made me think of a solution that would not be as limiting.

The width option did not merge well in the new system, and offered limited features compared to what is possible.
This solution will work on all  (out of many) parameters, instead of one and supports *all* features offered. (like setting width in terms of character width, or calculated).

This features would be added once and will work for most and new parameters without extra work, I see that as a win!

**Comment by DaveDavenport** [comment]
On hint of sdk: https://developer.mozilla.org/en-US/docs/Web/CSS/env()

**Reply by DaveDavenport** [reply]
This feature has been implemented. Needs testing.

```css
window {
  width: env( WIDTH, 30%);
}
```
